File: 0190-reverse-bits/0190-reverse-bits.py
class Solution:
    def reverseBits(self, n: int) -> int:
        result = 0
        
        for i in range(31, -1, -1):
            mask = n&1
            n>>=1
            result|=mask
            
            result<<=1
            
        return result>>1
            
            
            

File: 0190-reverse-bits/0190-reverse-bits.cpp
class Solution {
public:
    uint32_t reverseBits(uint32_t n) {
        int res = 0, pos = 31;
        
        for(int i=0;i<32;i++){
            res |= (n&1)<<pos;
            
            n>>=1;
            pos-=1;
        }
        
        return res;
    }
};